datalake/src/main/java/com/sequenceiq/datalake/flow/repair/handler/SdxRepairWaitHandler.java

```diff
@@ -0,0 +1,69 @@
+package com.sequenceiq.datalake.flow.repair.handler;
+
+import java.util.concurrent.TimeUnit;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.dyngr.exception.PollerException;
+import com.dyngr.exception.PollerStoppedException;
+import com.dyngr.exception.UserBreakException;
+import com.sequenceiq.cloudbreak.common.event.Selectable;
+import com.sequenceiq.datalake.flow.repair.event.SdxRepairFailedEvent;
+import com.sequenceiq.datalake.flow.repair.event.SdxRepairSuccessEvent;
+import com.sequenceiq.datalake.flow.repair.event.SdxRepairWaitRequest;
+import com.sequenceiq.datalake.service.sdx.PollingConfig;
+import com.sequenceiq.datalake.service.sdx.SdxRepairService;
+import com.sequenceiq.flow.reactor.api.handler.EventHandler;
+
+import reactor.bus.Event;
+import reactor.bus.EventBus;
+
+@Component
+public class SdxRepairWaitHandler implements EventHandler<SdxRepairWaitRequest> {
+
+    public static final int SLEEP_TIME_IN_SEC = 10;
+
+    public static final int DURATION_IN_MINUTES = 20;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SdxRepairWaitHandler.class);
+
+    @Inject
+    private EventBus eventBus;
+
+    @Inject
+    private SdxRepairService repairService;
+
+    @Override
+    public String selector() {
+        return "SdxRepairWaitRequest";
+    }
+
+    @Override
+    public void accept(Event<SdxRepairWaitRequest> event) {
+        SdxRepairWaitRequest stackDeletionWaitRequest = event.getData();
+        Long sdxId = stackDeletionWaitRequest.getResourceId();
+        String userId = stackDeletionWaitRequest.getUserId();
+        String requestId = stackDeletionWaitRequest.getRequestId();
```

Comment:
please add requestId to MDCContext with this: `MDCBuilder.addRequestIdToMdcContext(requestId);`